certora/specs/Gates.spec

```diff
@@ -107,31 +106,27 @@ rule cantReceiveShares(env e, method f, calldataarg args, address user) filtered
 rule cantSendShares(env e, method f, calldataarg args, address user, uint256 shares) filtered {
     f -> f.selector != sig:multicall(bytes[]).selector
 }{
-    require(currentContract.sharesGate != 0, "setup gating");
+    require (!canSendShares(user), "setup gating");
 
     // Trick to require that all the following addresses are different.
     require(MorphoMarketV1Adapter == 0x10, "ack");
     require(MorphoVaultV1Adapter == 0x11, "ack");
     require(currentContract == 0x12, "ack");
 
-    require (!canSendShares(user), "require that the user under scrutiny can't send shares");
-
     uint256 sharesBefore = balanceOf(user);
 
     f(e, args);
 
     assert balanceOf(user) >= sharesBefore;
 }
 
-// Check that transfers initiated from the vault, assuming the vault is no reentred, may only increase the balance of a given user when he can't send, and similarly the balance may only decrease when he can't receive.
+// Check that transfers initiated from the vault, assuming the vault is not reentred, may only increase the balance of a given user when he can't send, and similarly the balance may only decrease when he can't receive.
+// Assume that the vault only uses market v1 and vault v1 adapters, and that those are properly deployed to point to market v1 and vault v1 respectively.
```

Comment:
See [this thread](https://morpholabs.slack.com/archives/C02N7CZ088N/p1754914323459289) for more detail